lib/visual-website-optimizer/test.js

```diff
@@ -39,19 +39,71 @@ describe('Visual Website Optimizer', function(){
 
   describe('before loading', function(){
     beforeEach(function(){
+      analytics.stub(vwo, 'load');
       analytics.stub(vwo, 'replay');
+      analytics.stub(vwo, 'sendRoots');
     });
 
     describe('#initialize', function(){
       it('should call replay', function(){
         analytics.initialize();
         analytics.page();
-        analytics.called(vwo.replay);
+        tick(function(){
+          analytics.called(vwo.replay);
+          done();
+        });
+      });
+
+      it('should call #sendRoots if listen is enabled', function(done){
+        vwo.options.listen = true;
+        analytics.initialize();
+        analytics.page();
+        tick(function(){
+          analytics.called(vwo.sendRoots);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('#replay', function(){
+    beforeEach(function(){
+      analytics.stub(analytics, 'identify');
+    });
+
+    it('should replay experiments', function(done){
+      vwo.options.listen = true;
+      analytics.initialize();
+      analytics.page();
+
+      tick(function(){
+        window._vis_opt_queue[0]();
+        analytics.called(analytics.identify, { "Experiment: 1": 'Variation' });
+        done();
       });
     });
+  });
+
+  describe('#sendRoots', function(){
+    beforeEach(function(){
+      analytics.stub(analytics, 'track');
+    });
 
-    describe('#replay', function(){
-      it('should replay variation data');
+    it('should send active experiments', function(done){
+      vwo.options.listen = true;
+      analytics.initialize();
+      analytics.page();
+
+      tick(function(){
+        window._vis_opt_queue[1]();
+
+        analytics.called(analytics.track, 'Experiment Viewed', {
+          experimentId: "1",
+          variationName: 'Variation' },
+          { context: { integration: { name: 'visual-website-optimizer', version: '1.0.0' }}
```

Comment:
it's worth storing this name/version metadata on the integration proto so you can reference it here. Otherwise we're going to have to update these lines every time we bump the integration version.